build-logic/src/main/kotlin/org/jetbrains/conventions/base-java.gradle.kts

```diff
@@ -8,15 +8,24 @@ package org.jetbrains.conventions
  */
 
 plugins {
+    id("org.jetbrains.conventions.base")
     java
 }
 
 java {
     toolchain {
-        languageVersion.set(JavaLanguageVersion.of(8))
+        languageVersion.set(dokkaBuild.mainJavaVersion)
     }
 }
 
 java {
     withSourcesJar()
 }
+
+tasks.withType<Test>().configureEach {
+    useJUnitPlatform()
```

Comment:
https://github.com/Kotlin/dokka/pull/2918#issuecomment-1471893538

> > I think that link doesn't work, it just links to the whole PR. Which bit needs to be fixed?
> 
> Oh, sorry. In the now deleted `plugins/build.gradle.kts` there used to be configuration for `tasks.test` with `useJUnitPlatform()` - I think that's what made the tests run

Very good catch! `useJUnitPlatform()` should have been moved to a convention plugin in #2704, but it was missed.

Hopefully this fix 'just works'. Some subprojects might need to override this with `useJUnit()`.


> The only thing I didn't understand was how tests were run in other modules like core - found no configuration for tests whatsoever

I _guess_ that `useJUnit()` is the default, and most projects just use plain kotlin.test.

I think all tests should be updated to use JUnit 5.

Reply:
nm, looks like `useJUnitPlatform()` isn't a good fit, a lot of subprojects need it, but should be revisited later. I'll set `useJUnit()` as the default for now.

Reply:
Created an issue for longer term refactoring: #2924

For now it'd be good to just run all the tests as before

Reply:
I've added the necessary JUnit runtime dependencies so JUnit Platform can run both JUnit 4 and 5 tests https://junit.org/junit5/docs/current/user-guide/#running-tests-build-gradle-engines-configure

Locally I get more than 1130 test running